pkg/intermediate/aggregate.go

```diff
@@ -124,16 +130,14 @@ func (a *aggregation) correlateRecords(flowKey FlowKey, record entities.Record)
 				}
 			}
 		}
+		a.updateFlowKeyRecords(flowKey, existingRecords)
 	}
-	a.FlowKeyRecordLock.Unlock()
 	a.addRecordToMap(flowKey, record)
 	a.removeDuplicates(flowKey)
 }
 
 func (a *aggregation) removeDuplicates(flowKey FlowKey) {
-	a.FlowKeyRecordLock.Lock()
-	defer a.FlowKeyRecordLock.Unlock()
-	records := a.flowKeyRecordMap[flowKey]
+	records := a.GetFlowKeyRecordMap()[flowKey]
```

Comment:
This is still inside a lib function (i.e. not an external caller).
Do we need to make a copy of the map?
Can we just lock/unlock it?

Reply:
Sure. Updated. Thanks!